internal/probing/parse.go

```diff
@@ -0,0 +1,80 @@
+package probing
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+)
+
+// Parse takes a list of ObjectSetProbes commonly defined within a ObjectSet phase and
+// compiles a single Probe to test objects with.
+func Parse(ctx context.Context, packageProbes []corev1alpha1.ObjectSetProbe) (Prober, error) {
+	probeList := make(List, len(packageProbes))
+	for i, pkgProbe := range packageProbes {
+		probe := ParseProbes(ctx, pkgProbe.Probes)
+		var err error
+		probe, err = ParseSelector(ctx, pkgProbe.Selector, probe)
+		if err != nil {
+			return nil, fmt.Errorf("parsing selector of probe #%d: %w", i, err)
+		}
+		probeList[i] = probe
+	}
+	return probeList, nil
+}
+
+// ParseSelector reads a corev1alpha1.ProbeSelector and wraps a Prober, only executing it only when the selector matches.
```

Comment:
I am not sure what `only executing it only when the selector matches.` means here. I think it can be removed